x-pack/legacy/plugins/siem/public/components/links/index.tsx

```diff
@@ -117,34 +125,102 @@ export const CertificateFingerprintLink = React.memo<{
 
 CertificateFingerprintLink.displayName = 'CertificateFingerprintLink';
 
-export const ReputationLink = React.memo<{ children?: React.ReactNode; domain: string }>(
-  ({ children, domain }) => (
-    <EuiLink
-      href={`https://www.talosintelligence.com/reputation_center/lookup?search=${encodeURIComponent(
-        domain
-      )}`}
-      target="_blank"
-    >
-      {children ? children : domain}
-    </EuiLink>
-  )
-);
+enum DefaultReputationLink {
+  'virustotal.com' = 'virustotal.com',
+  'talosIntelligence.com' = 'talosIntelligence.com',
+}
+
+export interface ReputationLinkSetting {
+  name: string;
+  url_template: string;
+}
+
+function isDefaultReputationLink(name: string): name is DefaultReputationLink {
+  return (
+    name === DefaultReputationLink['virustotal.com'] ||
+    name === DefaultReputationLink['talosIntelligence.com']
+  );
+}
+const isReputationLink = (
+  rowItem: string | ReputationLinkSetting
+): rowItem is ReputationLinkSetting =>
+  (rowItem as ReputationLinkSetting).url_template !== undefined &&
+  (rowItem as ReputationLinkSetting).name !== undefined;
+
+export const DEFAULT_NUMBER_OF_REPUTATION_LINK = 5;
+const ReputationLinkComponent: React.FC<{
+  overflowIndexStart?: number;
+  allItemsLimit?: number;
+  showDomain?: boolean;
+  domain: string;
+}> = ({
+  overflowIndexStart = DEFAULT_NUMBER_OF_REPUTATION_LINK,
+  allItemsLimit = DEFAULT_NUMBER_OF_REPUTATION_LINK,
+  showDomain = false,
+  domain,
+}) => {
+  const [ipReputationLinksSetting] = useUiSetting$<ReputationLinkSetting[]>(
+    IP_REPUTATION_LINKS_SETTING
+  );
+  const [ipReputationLinks, setIpReputationLinks] = useState<ReputationLinkSetting[]>(
+    ipReputationLinksSetting
+  );
+
+  const defaultNameMapping: Record<DefaultReputationLink, string> = useMemo(
+    () => ({
+      [DefaultReputationLink['virustotal.com']]: i18n.VIEW_VIRUS_TOTAL,
+      [DefaultReputationLink['talosIntelligence.com']]: i18n.VIEW_TALOS_INTELLIGENCE,
+    }),
+    []
+  );
+
+  useEffect(() => {
+    setIpReputationLinks(
+      ipReputationLinks
+        ?.slice(0, allItemsLimit)
+        .filter(({ url_template }) => !isUrlInvalid(url_template))
+        .map(({ name, url_template }: { name: string; url_template: string }) => {
+          return {
+            name: isDefaultReputationLink(name) ? defaultNameMapping[name] : name,
+            url_template: url_template.replace(`{{ip}}`, encodeURIComponent(domain)),
+          };
+        })
+    );
+  }, [domain, overflowIndexStart, setIpReputationLinks, defaultNameMapping]);
```

Comment:
Consider listing `ipReputationLinks` as a dependency here, because it's mutable state

Reply:
I added that before but ended up my component kept updating so I removed it, not sure it there's any better way to handle this...

Reply:
Per our offline discussion, since there is nothing really "effectful" happening this `useEffect`, the projected/derived value of `ipReputationLinks` can be computed inside a `useMemo` instead of a `useEffect`. This would eliminate the need to update `ipReputationLinks` via the `setIpReputationLinks` function provided by `useState`. (`useState` would no longer be necessary).